ksp_plugin/vessel.hpp

```diff
@@ -83,9 +83,15 @@ class Vessel {
   // Returns the part with the given ID.  Such a part must have been added using
   // |AddPart|.
   virtual not_null<Part*> part(PartId id) const;
+  virtual std::map<PartId, not_null<std::unique_ptr<Part>>> const& parts()
+      const;
```

Comment:
There is a bit of an abstraction leak here as we expose the data structure internal to the vessel.  Maybe it would be better to have `GetSomePart` (returning the first one, but that's not part of the contract) and `DoForAllParts` (taking a lambda).